### PR TITLE
Catch the exception raised in try block and send via error in callback

### PR DESCRIPTION
**Pull request by faisal-saleh**
Reason: When an exception is raised in AsyncStorageModule, App is crashing.

### CODE REVIEW

**Comment by rozele** [comment]
I'm concerned about masking the real problem here. Why is this exception occurring in the first place?

**Comment by faisal-saleh** [comment]
The actual message of the exception is 'Unable to remove the file to be replaced'. Perhaps a previous read/write operation is not releasing the file fast enough before a new operation starts. We noticed the exception is raised frequently when AsyncStorage is used fetch the stored values very quickly (due to user interaction).

**Comment by matthargett** [comment]
Turning a native crash into a recoverable JS promise rejection seems like a good tradeoff to me, and not necessarily "masking the real problem". I do agree that there's this DX improvement and a separate issue of the race, the latter of which needs to be investigated and fixed in a separate PR, though.

**Comment by UsmanAli-10p** [comment]
@rozele any update on this PR?

File: ReactWindows/ReactNative.Shared/Modules/Storage/AsyncStorageHelpers.cs [hunk 1]
```diff
@@ -157,5 +158,14 @@ public static JObject GetInvalidValueError(string key)
         {
             return GetError(key, "Invalid Value");
         }
+
+        public static JObject GetError(Exception ex)
+        {
+            return new JObject
+            {
+                { "message", ex.Message },
+                { "key", ex.GetType().Name },
```
Comment (matthargett) [inline ReactWindows/ReactNative.Shared/Modules/Storage/AsyncStorageHelpers.cs]: would it be possible to stash the stack trace in here as well? then the JS promise rejection would have more remediation information in it.

File: ReactWindows/ReactNative.Shared/Modules/Storage/AsyncStorageHelpers.cs [hunk 2]
```diff
@@ -157,5 +158,15 @@ public static JObject GetInvalidValueError(string key)
...
```
Comment (psivaram) [inline ReactWindows/ReactNative.Shared/Modules/Storage/AsyncStorageHelpers.cs]: Can you also stash ex.HResult. It is sometimes embedded in the message, but sometimes it is not. Having a HResult makes debugging easy, especially if the bug report is from a machine having a different user locale than what the dev understands

**Comment by rozele** [comment]
Closing - another PR came in with UWP fix as well.

**Pull request closed without merging**